vital/types/rotation.cxx

```diff
@@ -151,12 +149,11 @@ void
 rotation_< T >
 ::get_yaw_pitch_roll( T& yaw, T& pitch, T& roll ) const
 {
-  Eigen::Matrix< T, 3, 3 > rotM( this->matrix() );
-  T cos_p = T( std::sqrt( double( rotM( 1, 2 ) * rotM( 1, 2 ) ) + rotM( 2, 2 ) * rotM( 2, 2 ) ) );
-
-  yaw   = T( std::atan2( double( rotM( 0, 0 ) ), double( rotM( 0, 1 ) ) ) );
-  pitch = T( std::atan2( double( rotM( 0, 2 ) ), double(cos_p) ) );
-  roll  = T( std::atan2( double( -rotM( 1, 2 ) ), double( -rotM( 2, 2 ) ) ) );
+  roll = std::atan2( 2.0 * ( q_.w() * q_.x() + q_.y() * q_.z() ),
```

Comment:
Is this supposed to produce equivalent results or has there been a change of convention for the input data?

Reply:
The object's internal representation has changed convention, so `this` is different. I also switch from the `quaternion -> matrix -> ypr` method shown previously to a more direct / efficient `quaternion -> ypr` method.

Reply:
I see. So in both cases we're just converting to/from NED YPR to quaternions?

Reply:
Yes.